flake.nix

```diff
@@ -442,7 +442,10 @@
               constituents =
                 lib.collect lib.isDerivation packages.tests;
             };
-            ci.benchmarks = packages.benchmarks.cardano-wallet // {
+
+            ci.benchmarks =
+            packages.benchmarks.cardano-wallet-benchmarks //
+            packages.benchmarks.cardano-wallet // {
```

Comment:
This makes a little sense from the outside, is it a temporary state of the things?

Reply:
No idea, how should it be done ?

Reply:
In general, the nix expression `packages.benchmarks.cardano-package-name` contains the benchmark components of the cabal package `cardano-package-name`. In this case, we have two cabal packages: `cardano-wallet` and `cardano-wallet-benchmarks`.

> Why to have wallet benchmarks twice?
> What's the difference between `benchmarks.cardano-wallet-benchmarks` and `benchmarks.cardano-wallet`?

I have put the benchmarks in `cardano-wallet-benchmarks` into a separate packages because they do not have a *compile-time* dependency on `cardano-wallet`, only a *run-time* dependency.

I'm not against putting them in the same `cardano-wallet.cabal` file, but I felt that putting the run-time benchmarks into a separate package would be better for enforcing the phase separation (compile/run).

Reply:
Ok, now I see your logic. This is more like a naming issue. I wonder if we can have this distinction "compile/runtime" reflected in the nix derivation names?